puppet/site.pp

```diff
@@ -10,97 +14,70 @@
   ensure => present
 } ->
 package {"build-essential":
   ensure => present
 } ->
-# include update_package_manager
 ### END nokogiri deps
 
-apt::ppa{'ppa:brightbox/ruby-ng':}
-
-package { 'ruby-switch':
-  ensure => present,
-} ->
-package { 'ruby2.6':
-  ensure => present,
-  require => [ Class['apt'], Apt::Ppa['ppa:brightbox/ruby-ng'] ]
-} ->
-package { 'ruby2.6-dev':
-  ensure => present,
-  require => [ Class['apt'], Apt::Ppa['ppa:brightbox/ruby-ng'] ]
-} ->
-
-exec { 'set ruby':
-  command => 'ruby-switch --set ruby2.6',
-  path => ['/usr/bin'],
-  require => Package['ruby-switch']
-} ->
-
-exec { 'bundler':
-  command => 'gem install bundler',
-  path => ['/usr/bin']
-} ->
-
-# update rubygems and install application gems
-exec { 'install rubygems update':
-  command => 'gem install rubygems-update',
-  path    => ['/usr/local/bin','/usr/bin', '/bin'],
-  user    => 'root',
-  group   => 'root',
-  require => [ Exec['bundler'] ]
-} ->
-
-exec { 'update rubygems':
-  command => 'update_rubygems',
-  path    => ['/usr/local/bin','/usr/bin', '/bin'],
-  user    => 'root',
-  group   => 'root'
-} ->
-
-exec { 'gem update':
-  command => 'gem update --system',
-  path    => ['/usr/local/bin','/usr/bin', '/bin'],
-  user    => 'root',
-  group   => 'root'
+class { 'rbenv':
+  install_dir => '/home/vagrant/rbenv',
+  owner => 'vagrant',
+  group => 'vagrant',
 }
-
-if $environment == 'ci' {
-  package { 'rake':
-    provider => 'gem',
-    ensure   => 'installed'
-  }
+-> exec { 'rbenv-build-prepare-git':
+  command => 'git config --global --add safe.directory /home/vagrant/rbenv/plugins/ruby-build',
+  path => ['/usr/local/bin', '/usr/bin', '/bin'],
+  environment => ['HOME=/home/vagrant'],
+}
+-> rbenv::plugin { 'rbenv/ruby-build': }
+-> rbenv::build { $ruby_ver:
+  bundler_version => $bundler_ver,
+  owner => 'vagrant',
+  group => 'vagrant',
+  global => true,
+}
+-> rbenv::gem { 'builder': ruby_version => $ruby_ver }
+-> exec { 'gem_update':
+  command => "gem update --system ${rubygems_ver}",
+  path    => ['/home/vagrant/rbenv/shims', '/usr/local/bin','/usr/bin', '/bin'],
 }
 
 unless $environment == 'ci' {
   # dep for geos gems
   package {"libgeos-dev":
     ensure => present,
-    require => Exec['bundler']
+    require => Exec['gem_update']
   }
 
   # install application gems
   exec { 'do_bundle_install':
-    cwd => '/vagrant',
-    environment => 'HOME=/vagrant',
-    command => 'bundle install',
-    path => ['/usr/local/bin', '/usr/bin', '/bin'],
+    cwd     => "${app_root}",
+    environment => "HOME=${app_root}",
+    command => "bundle _${bundler_ver}_ install",
+    path => ['/home/vagrant/rbenv/shims', '/usr/local/bin','/usr/bin', '/bin'],
     user => 'vagrant',
     group => 'vagrant',
-    require => [ Exec['bundler'] ]
+    require => [ Exec['gem_update'] ]
+  }
+
+  class { "nsidc_solr":
+    jetty_host => '0.0.0.0'
   }
 
-  # nokogiri 'build native' dep
-  package { 'zlib1g-dev':
+  file_line { "solr_modules":
```

Comment:
Solr 9 doesn't include the "scripting" module by default anymore for some reason.  To get it to load, when the Solr instance is started, the docs say to either have it as an environment variable or as a system property.  This approach is making it an environment variable.